### DISCUSSION TITLE
inconsistence behaviour with different keymaps

### DISCUSSION THREAD

**Discussion by spiderbit** [stumpwm/stumpwm#1168] [Announcements]
I did a hack to either open a emacs frame with emacsclient and forward a keycombination or if I already have a emacs window focused send the keys directly to it:
```lisp
(defcommand emacs-start-map-c () ()
	    (if (equal "Emacs" (window-class (current-window)))
		(run-commands "exec xdotool key ctrl+space c")
		(run-commands "exec emacsclient -c -e '(setq unread-command-events (nconc (listify-key-sequence(kbd \"C-SPC c\"))))'")))
```
now I bound it to the c and w key in the root-map, so you would expect both operate the same way:
```lisp
(undefine-key *root-map* (kbd "c"))
(define-key *root-map* (kbd "c") "emacs-start-map-c")
(undefine-key *root-map* (kbd "w"))
(define-key *root-map* (kbd "w") "emacs-start-map-c")
```
but funny enough it does not, when I am in a emacs window with the w key I get the full Ctrl-space c combination executed and it waits for my next letter, with the "c key" it omits the c letter in this case.

I tried restart soft/hard logged out of stumpwm but it stays the same. Why can there be different behaviour based on the key binding used to execute it?

**Comment by dmb2** [comment]
Hi,

You have many layers of synthesized events going on.  What is the end goal of what you want to do? I think:
C-t c should always run "C-SPC c" in an emacsclient window.

Can you use `send-fake-key` and `run-or-raise`?

**Comment by spiderbit** [comment]
Well I want to use stumpwm my window-manager or have and get it as close as possible to my former exwm solution, where I can type directly emacs commands on the normal modifier and I don't want to map each single key binding again like copying it.

Because if I change my emacs config I don't want to have to change on top of that my stumpwm config, so therefor I want to just preexecute either check if I have focused a emacs screen or potentianally create such "frame" and then send the at least last key based on what I type just forwarding it.

So that I can have let's say Ctrl-Space w + 20 different keys to just do what this keys are mapped to in emacs and that for like 10 or 20 different key-maps.

Some keys only have 2 keys like leader keys (Ctrl-space) + 1 more most have 2 more.

C-t is bound to transpose chars in emacs and you mean maybe in stumpwm but in stumpwm I have a different leader key...
